ignite/contrib/metrics/fid.py

```diff
@@ -0,0 +1,89 @@
+import os
+from typing import Callable, Sequence, Union
+
+import torch
+import torch.nn as nn
+from torchvision import transforms
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+try:
+    from scipy import linalg
+except ImportError:
+    raise RuntimeError("This contrib module requires scipy to be installed.")
+
+
+__all__ = ["FID"]
+
+
+class FID(Metric):
+    """Calculates FID metric
+
+    """
+
+    def __init__(
+        self, output_transform: Callable = lambda x: x, fid_model: nn.Module = None,
+    ):
+        if fid_model is None:
+            try:
+                from torchvision import models
+
+                fid_model = models.inception_v3(pretrained=True, transform_input=True)
+            except ImportError:
+                raise ValueError("Argument fid_model should be set")
+
+        self._device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
Device should be set with `super(FID, self).__init__(output_transform=output_transform, device=device)`